File: solutions/elixir/lasagna/3/lib/lasagna.ex
defmodule Lasagna do
  @lasagna_cook_time 40
  @layer_prep_time 2

  def expected_minutes_in_oven(), do: @lasagna_cook_time
  def remaining_minutes_in_oven(minutes), do: @lasagna_cook_time - minutes
  def preparation_time_in_minutes(layers), do: layers * @layer_prep_time
  def alarm(), do: "Ding!"

  def total_time_in_minutes(layers, elapsed_time),
    do: preparation_time_in_minutes(layers) + elapsed_time
end

File: solutions/java/lasagna/3/src/main/java/Lasagna.java
public class Lasagna {
    private static final int EXPECTED_MINUTES_IN_OVEN = 40;
    private static final int PREPARATION_TIME_PER_LAYER = 2;

    public int expectedMinutesInOven() {
        return EXPECTED_MINUTES_IN_OVEN;
    }

    public int remainingMinutesInOven(int bakeTime) {
        return expectedMinutesInOven() - bakeTime;
    }

    public int preparationTimeInMinutes(int layers) {
        return PREPARATION_TIME_PER_LAYER * layers;
    }

    public int totalTimeInMinutes(int layers, int actualMinutes) {
        return preparationTimeInMinutes(layers) + actualMinutes;
    }
}